test/e2e/end-to-end.test.js

```diff
@@ -331,6 +337,9 @@ describe('test video api', () => {
       expect(segment2.tags[0].rank).toBe(11);
       expect(segment2.tags[0].tag.name).toBe('fear');
       expect(segment2.tags[0].tag._id).toBe(segment1.tags[0].tag._id);
+
+      expect(removedTag.segments.length).toBe(0);
+      expect(removedTag.segmentCount).toBe(0);
```

Comment:
Note that this is the test that currently fails.

Reply:
Updated segment.model.js to update tag counts for both old (potentially deleted) and new segments.  All end to end tests are now passing.